t/14-grutasks.t

```diff
@@ -608,6 +608,38 @@ subtest 'download assets with correct permissions' => sub {
     ok -f $assetpath, 'asset downloaded';
 };
 
+subtest 'finalize job results' => sub {
+    my %settings = (
+        DISTRI  => 'Unicorn',
+        FLAVOR  => 'pink',
+        VERSION => '42',
+        BUILD   => '666',
+        ISO     => 'whatever.iso',
+        MACHINE => "RainbowPC",
+        ARCH    => 'x86_64',
+        TEST    => 'minion',
+    );
+
+    my $job = $schema->resultset('Jobs')->create_from_settings(\%settings);
+    $job->discard_changes;
+    $job->insert_module({name => 'a', category => 'a', script => 'a', flags => {}});
+    $job->update_module('a', {result => 'ok', details => [{title => 'wait_serial', text => 'a-0.txt'}]});
+    $job->insert_module({name => 'b', category => 'b', script => 'b', flags => {}});
+    $job->update_module('b', {result => 'fail', details => [{title => 'wait_serial', text => 'b-0.txt'}]});
+    $job->update;
+    $job->discard_changes;
+    path($job->result_dir, 'a-0.txt')->spurt('Foo');
+    path($job->result_dir, 'b-0.txt')->spurt('Bar');
+    $job->done;
+    $job->discard_changes;
+    $t->app->minion->perform_jobs;
+    is($job->result, OpenQA::Jobs::Constants::FAILED, 'job result is failed');
+    my @modlist = $job->modules;
+    is($modlist[0]->results->{details}->[0]->{text_data}, 'Foo');
+    is($modlist[1]->results->{details}->[0]->{text_data}, 'Bar');
```

Comment:
Of course non of these checks assert whether the `text_data` now comes from the JSON file. To keep things simple you could just remove the `*.txt` files before the checks. If they are really no longer read from this shouldn't be a problem (in the context of this test).

Reply:
`text_data` has nowhere else to come from at the moment. `JobModules::results()` does not load it from external files. I'll add a second minion job in the next commit that'll delete the obsolete files.

Reply:
Ok, sounds good enough.